### DISCUSSION TITLE
Cross-browser DAV sync disrupted by failing Android app + new features. I am confused..

### DISCUSSION THREAD

**Discussion by bolohori** [floccusaddon/floccus#1525] [Q&A]
###My gear and setup

I use my Cloud Storage drive webDAV server to sync my bookmarks folder between 6 browsers on Window 11.  

**Auto-sync** on with **20min-25min** intervals  worked even with 2 or 3 browsers opened simultaneously.  For the past 6 months, it kept 
~19k links in perfect **sync** in the same **bookmarks-sync.xbel file** . 

_**merge local storage**_ mode on the most used:
- **Microsoft Edge** - Version 122.0.2348.0 (Official build) dev (64-bit) - flocus **5.0.6**
- **Vivaldi** - Version 122.0.2325.0 - floccus **5.0.6** 
- **Firefox** - 122.0b3 (64-bit) - not used this time
_**undo local changes**_ mode on rare occasions for
- **Opera, Librefox & Chrome** - only with upload but not used  this time

###Settings and first dilemma 

My webDAV url is **https://storage.example.com/dav** (443 port) and it has a root folder named "Cloud Storage"

**webDAV url:**  https://storage.example.com/dav/Cloud%20Storage/my-bookmarks/floccus/
**_Bookmarks file path:_** bookmarks-sync.xbel

- took a few attempts because the field description from the _Bookmarks file path:_  (_a path to the bookmarks file relative to your WebDAV URL (all folders in the path must already exist). e.g. personal_stuff/bookmarks.xbel_)  gives a very misleading example, and there is no visible mention or input validation that helps the user understand where the dash should be placed (after the url, or before the file path) .  ( this misleading design pattern was also mentioned in [this issue](https://github.com/floccusaddon/floccus/issues/1263#issue-1399008412)

 ##### **Working in all browsers mentioned** 
 
### First big problem

Last week I cleaned my phone and organized apps and thought it would be nice to have my bookmarks at my fingertip 😊

So I installed floccus **5.0.7** on my phone with**Oxygen OS 11.1.2.2**  (**Android 11 4.9.227-perf+**)

TL;DR phone failure drama
```markdown
- **the same** path for the path values working on desktop with both _**merge changes**_ and _**undo local changes**_ - the app froze
- same config and *merge local changes*  - the app frozen, 
- no progress indication, failure error, or other warning.
- had to force stop and restart it 
- even after clearing app cache, the sync arrows were still blue and it stalled
- I even tried to create a  root folder "Favorites Bar" and add 1 bookmark then rinse and repeat. Same thing happened. 
- I even waited for1h while the app was frozen and nothing happened, not a single bookmark was uploaded and I was in the blind, 

When I changed it to 

**webDAV url:**  https://storage.example.com/dav/
**_Bookmarks file path:_** /Cloud%20Storage/my-bookmarks/floccus/bookmarks-sync.xbel

- the app had different behavior, didn't freeze anymore, but the sync icon was blue and spinning 
- 
```
#### Now here is where the little nightmare starts
I want to share the resources on a directory website, so I need to move most of them on my self-hosted server.

This week I saved and reorganized around 100 bookmarks in Vivaldi. During the weekend I managed to clean and reorganize ~4k in MS Edge and never worried about the sync status, because I saw the blue icon and ok sign and I was almost certain the changes were already synced on Edge.

Only when I finished with Edge, I noticed it had Scheduling sync status and disabled buttons, with no other browser or the app running. Last sync was 4 days ago and the Vivaldi changes weren’t synced either…so I quickly exported a copy and started Vivaldi, which also had the same issue. Both logs saying the “bookmarks-sync.xbel.lock” is locked, try again soon.

After investigating, I learned about the new insta sync on browser start and the Scheduling features in the new release.

I tried over and over for hours being stuck with that Scheduling status and no other message or at least indicator that something is working.  
All the small design issues were slowly creeping in and made everything even more frustrating. 

 - I removed the profile from the phone, closed Edge, started Vivaldi, managed to make a sync.
- At this point, I wasn’t even sure the password was being saved between 2 settings saves, because the input is empty when reopening settings. 
- The auto-sync button was also removed the popup interface, and with the new 1 second sync feature triggering, it makes it harder to stop sync on a browser before corrupting the .lock file.

````markdown 
- At some point I tried to create a profile that uploads a html file from Edge to a new location in the Cloud ...  I figured that would be a different job , but they ended up being locked on Scheduled on all instances, with no working sync job. 
- this pointed out another issue, where a name of server the path is not showed in the sync task cards from the extension Popup UI. So if you sync the same local folder in 2 different locations, you can't tell which card is for what path without going in the settings. Sometimes I noticed they even changed their positions, creating more confusion
```
 - I finally deleted the .lock file on my webDAV and restarted Vivaldi to merge the small changes before reopening Edge for the big merge. 
 
 ##### I finally saw a sync complete message after long frustrating hours…but only in Vivaldi
-  after restarting Edge, I managed to make it work again, but I ended up with the 4k bookmarks back in it. .. 

I still have the html export file with the work I done in Edge. A new sync should add ~100 links from the existing version to the backup file and then delete 4k coming from  the backup...  I should ask for instructions on how to do it, but I am not ready to go again .... 

I am a frontend developer and the lesson I learned after this experience, is that if I want better OSS apps, I'd better step-up my game, learn how to contribute myself and  start writing my ideas for better features.

If you reached this far, thank you for reading and sorry for the long post. 

Cheers

**Comment by bolohori** [comment]
Solved by using clean syncs in different folders for both files then used the lil amazing extension Bookmark Dupes to clean everything.

**Comment by marcelklehr** [comment]
> webDAV url: [storage.example.com/dav/Cloud%20Storage/my-bookmarks/floccus](https://storage.example.com/dav/Cloud%20Storage/my-bookmarks/floccus/)
Bookmarks file path: bookmarks-sync.xbel

This seems like you're missing a slash at the end of the URL.

This post made me realize that Edge is still a version behind on releases.

No idea what's going on with the app when it froze only in a certain configuration :/

**Comment by marcelklehr** [comment]
> this misleading design pattern was also mentioned

The issue you mentioned was a problem in Owncloud in the end, are you sure that's the issue you wanted to link?
